pandas/core/arrays/string_arrow.py

```diff
@@ -130,18 +130,22 @@ class ArrowStringArray(ObjectStringArrayMixin, ArrowExtensionArray, BaseStringAr
 
     def __init__(self, values) -> None:
         _chk_pyarrow_available()
-        if isinstance(values, (pa.Array, pa.ChunkedArray)) and pa.types.is_string(
-            values.type
+        if isinstance(values, (pa.Array, pa.ChunkedArray)) and (
+            pa.types.is_string(values.type)
```

Comment:
Is there a reason why here we only test for string but in the dictionary-encoded case we allow both string and large_string?

Reply:
This part of the diff went away now that https://github.com/pandas-dev/pandas/pull/59479 is merged. But to answer your question: the reason is that we are checking the types here for which we are going to cast to `large_string`. 
If the input is already `large_string`, we don't need to cast (although the cast would be cheap zero copy anyway, I think), only if it is `string` or `dictionary` (of either string type) we need to cast, to ensure the final array we store is always of arrow type `large_string`.